components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/utils/AppGatewayAdminClient.java

```diff
@@ -0,0 +1,410 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.appmgt.impl.utils;
+
+import org.apache.axiom.om.OMElement;
+import org.apache.axis2.AxisFault;
+import org.apache.commons.lang.StringUtils;
+import org.wso2.carbon.appmgt.api.AppManagementException;
+import org.wso2.carbon.appmgt.api.model.APIIdentifier;
+import org.wso2.carbon.appmgt.gateway.dto.stub.WebAppData;
+import org.wso2.carbon.appmgt.gateway.stub.AppGatewayAdminAppManagementExceptionException;
+import org.wso2.carbon.appmgt.gateway.stub.AppGatewayAdminStub;
+import org.wso2.carbon.appmgt.impl.dto.Environment;
+import org.wso2.carbon.appmgt.impl.template.APITemplateBuilder;
+import org.wso2.carbon.appmgt.impl.template.APITemplateException;
+import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
+
+import javax.xml.stream.XMLStreamException;
+import java.io.StringWriter;
+import java.rmi.RemoteException;
+
+public class AppGatewayAdminClient extends AbstractAPIGatewayAdminClient {
```

Comment:
Fixed in https://github.com/wso2/carbon-appmgt/pull/539/commits/207d3b1c14192fee91f6d02b39c54f42bdf46e92